src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -110,6 +110,33 @@ public function remove(string $key)
         unset($this->parameters[$key]);
     }
 
+    /**
+     * Checks whether the given parameter matches the given regular expression.
+     *
+     * @return bool Returns true if the paramter matches the regular expression.
+     *              However, false is returned if the parameter is either not defined or if the parameter did not match the regular expression.
+     *
+     * @throws \InvalidArgumentException if regexp is not a valid regular expression
+     */
+    public function match(string $key, string $regexp): bool
+    {
+        if (!$this->has($key)) {
+            return false;
+        }
+
+        if (\strlen($regexp) < 2 || ('/' !== $regexp[0] || '/' !== $regexp[\strlen($regexp) - 1])) {
+            throw new \InvalidArgumentException('Parameter "regexp" is not a valid regular expression.');
+        }
+
+        $match = preg_match($regexp, $this->getString($key));
+
+        if (false === $match) {
+            throw new \InvalidArgumentException('Parameter "regexp" is not a valid regular expression.');
```

Comment:
It could be nice to show the message returned by https://www.php.net/manual/fr/function.preg-last-error-msg.php

Reply:
Do you mean instead of `'Parameter "regexp" is not a valid regular expression.'`?